src/status_im/extensions/core.cljs

```diff
@@ -245,6 +246,15 @@
      (when-let [command (last (first (filter #(= (ffirst %) (name hook-id)) (:id->command db))))]
        (commands-sending/send cofx current-chat-id command params)))))
 
+(handlers/register-handler-fx
+ :extensions/show-selection-screen
+ (fn [cofx [_ _ {:keys [on-select] :as params}]]
+   (navigation/navigate-to-cofx cofx
+                                :selection-modal-screen
+                                (assoc params :on-select #(do
+                                                            (re-frame/dispatch [:navigate-back])
+                                                            (re-frame/dispatch (on-select %)))))))
```

Comment:
let's make this a single atomic event

Reply:
i think for such cases it's fine to use two events